libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -578,6 +616,71 @@ bool AP_BattMonitor::reset_remaining(uint16_t battery_mask, float percentage)
     return ret;
 }
 
+// Logs smart battery messages
+void AP_BattMonitor::log_smart_battery_info() const
+{    
+    for(uint8_t i = 0; i < _num_instances; i++) {
+        if (get_type(i) > AP_BattMonitor::Type::ANALOG_VOLTAGE_AND_CURRENT) {
```

Comment:
Urkh, no, please no.

Check the types explicitly, don't assume a range in an enumeration.

```suggestion
        if (get_type(i) == AP_BattMonitor::Type::ANALOG_VOLTAGE_AND_CURRENT) {
            continue;
        }
        if (get_type(i) == AP_BattMonitor::Type::BORING) {
            continue;
        }
```

Reply:
.... even better, just call `log_smart_battery_info()` on the backend.

Backends should log themselves....

Reply:
I noticed the SMBUS stuff, when moving logging to the new file, and am currently moving it to the backend of SMBUS with all of the piping in between. :)

Reply:
Moved to logging to backend.
Also explicitly checking the ENUM  now where I was doing the same thing in GCS_Common.